### PR TITLE
Replace components xml file by json file (issue #238)

### PR DESCRIPTION
**Pull request by tadam50**
Signed-off-by: Thomas ADAM <tadam@silicom.fr>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
'#238'

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
Using JAXB dependency

**What is the new behavior (if this is a feature change)?**
No JAXB dependency needed

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: single-line-diagram-core/src/test/java/com/powsybl/sld/library/ComponentsTest.java [hunk 1]
```diff
@@ -27,27 +25,38 @@
...
+                "\"subComponents\" : [ {",
+                    "\"name\" : \"BREAKER\",",
+                    "\"fileName\" : \"breaker.svg\",",
+                    "\"styleClass\" : null",
+                "} ],",
+                "\"styleClass\" : \"sld-breaker\"",
+                "} ]",
+                "}");
+
         Components components;
```
Comment (flo-dup) [inline single-line-diagram-core/src/test/java/com/powsybl/sld/library/ComponentsTest.java:56]: Very minor but you can merge that line with the line under

**Review by flo-dup — CHANGES_REQUESTED** [review]
Great, just a few minor comments

File: single-line-diagram-core/src/main/java/com/powsybl/sld/library/Component.java [hunk 2]
```diff
@@ -16,6 +19,11 @@
 
     private ComponentMetadata metadata;
 
+    @JsonCreator
+    public Component(@JsonProperty("") ComponentMetadata metadata) {
```
[the remark this answers is not in the document]
Comment (tadam50) [reply single-line-diagram-core/src/main/java/com/powsybl/sld/library/Component.java:23]: Do you want to keep Component class and remove ComponentMetadata class ?

[the remark this answers is not in the document]
Comment (flo-dup) [reply single-line-diagram-core/src/main/java/com/powsybl/sld/library/Component.java:23]: I'd rather keep Component class and remove ComponentMetadata.

**Review by flo-dup — APPROVED** [review]

**Pull request merged**